README.md

```diff
@@ -125,6 +125,69 @@ node {
 ```
 In the future we might migrate to a [BuildStep](http://javadoc.jenkins-ci.org/hudson/tasks/BuildStep.html) instead of a BuildWrapper.
 
+After integration of the [hashicorp-vault-pipeline-plugin](https://github.com/jenkinsci/hashicorp-vault-pipeline-plugin) this is now also possible
+##### Using global vault configuration
+
+```
+pipeline {
+    agent any
+    environment {
+        SECRET = vault path: 'secrets', key: 'username'
+    }
+    stages {
+        stage("read vault key") {
+            steps {
+                echo "${SECRET}"
+            }
+        }
+    }
+}
+```
+
+##### Using pipeline specific configuration
+
+```
+pipeline {
+    agent any
+    environment {
+        SECRET = vault path: 'secrets', key: 'username', vaultUrl: 'https://my-vault.com:8200', credentialsId: 'my-creds', engineVersion: 2
+    }
+    stages {
+        stage("read vault key") {
+            steps {
+                echo "${SECRET}"
+            }
+        }
+    }
+}
+```
+
+##### Masking secrets in console output
```

Comment:
In azure-keyvault plugin I used a block scoped step not the environment option and secrets were hidden fine:
https://github.com/jenkinsci/azure-keyvault-plugin#declarative